### PR TITLE
automated mm ratings and skill verification

### PR DESCRIPTION
**Pull request by bountyC0d3r**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    "q": "^1.5.1",
+    "superagent": "^8.0.3",
+    "superagent-prefix": "0.0.2",
+    "tc-core-library-js": "appirio-tech/tc-core-library-js.git#v2.6.4",
+    "topcoder-healthcheck-dropin": "^1.0.3",
+    "winston": "^3.8.2",
+    "xml2js": "^0.4.23"
+  },
+  "engines": {
+    "node": "10.x"
```
Comment (rakibansary) [inline package.json:29]: We should use the latest LTS

**Comment by bountyC0d3r** [comment]
using the latest LTS would be a problem, as somehow `ifxnjs` is giving
compilation error

File: src/common/helper.js [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    return content
+  }
+
+  return null
+}
+
+/**
+ * Function to fetch all the submissions for a given challenge
+ * @param {string} challengeId challengeId
+ * @returns {array} submissions
```
Comment (rakibansary) [inline src/common/helper.js]: Let's update @returns {array} submissions to `@return {Promise<array>}` - causes issues with linters reporting consumers of this method as using "unnecessary await"

File: src/common/helper.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  })
+
+  return latestSubmissions
+}
+
+
+/**
+ * Helper function returning prepared superagent instance for using with v5 challenge API.
+ * @param {String} token M2M token value
+ * @returns {Object} superagent instance configured with Authorization header and API url prefix
```
Comment (rakibansary) [inline src/common/helper.js]: `@returns {Promise<Object>}`

File: src/common/informixDB.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  ';DB_LOCALE=' +
+  config.get('INFORMIX.DB_LOCALE') +
+  ';UID=' +
+  config.get('INFORMIX.USER') +
+  ';PWD=' +
+  config.get('INFORMIX.PASSWORD')
+
+/**
+ * Get Informix connection using the configured parameters
+ * @return {Object} Informix connection
```
Comment (rakibansary) [inline src/common/informixDB.js]: `@returns {Promise<Object>}`

File: src/common/informixDB.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  } finally {
+    await informixSession.closeSync();
+  }
+
+}
+
+/**
+ * Get entries of `long_comp_result` table for the given `round_id`
+ * @param {string} roundId round id
+ * @returns {array} rows from long_comp_result table
```
Comment (rakibansary) [inline src/common/informixDB.js]: @returns {Promise<array>}

File: src/common/helper.js (hunk 2 above)
Comment (rakibansary) [reply src/common/helper.js]: image

**Review by rakibansary — CHANGES_REQUESTED** [review]

**Review by rakibansary — APPROVED** [review]

**Pull request merged**